pitest/src/test/java/org/pitest/mutationtest/engine/gregor/mutators/DSLMethodCallMutatorTest.java

```diff
@@ -0,0 +1,176 @@
+
+public class DSLMethodCallMutatorTest extends MutatorTestBase {
+
+	private static class HasVoidMethodCall implements Callable<String> {
+
+		private String s = "";
+
+		public void set(final int i) {
+			this.s = this.s + i;
+		}
+
+		public String call() throws Exception {
+			set(1);
+			return this.s;
+		}
+
+	}
+
+	static class HasIntMethodCall implements Callable<String> {
+
+		private static int i = 0;
+
+		public int set(final int newVal) {
+			i = newVal;
+			return i + 42;
+		}
+
+		@SuppressWarnings("finally")
+		public String call() throws Exception {
+			int c = 2;
+			try {
+				c = set(1);
+			} finally {
+				return "" + c;
+			}
+		}
+
+	}
+
+	static class HasDslMethodCall implements Callable<String> {
+
+		private int i = 0;
+
+		public HasDslMethodCall chain(final int newVal) {
+			this.i += newVal;
+			return this;
+		}
+
+		public void voidNonDsl(final int newVal) {
+			this.i += newVal;
+		}
+
+		public int nonDsl(final int newVal) {
+			this.i += newVal;
+			return i;
+		}
+
+		public String call() throws Exception {
+			HasDslMethodCall dsl = this;
+			dsl.chain(1).nonDsl(3);
+			return "" + dsl;
+		}
+
+		@Override
+		public String toString() {
+			return "HasDslMethodCall [i=" + i + "]";
+		}
+
+	}
+
+	static class ParentHasDslMethodCall<X extends ParentHasDslMethodCall>   {
+
+		private int i = 0;
+
+		public X chain(final int newVal) {
+			this.i += newVal;
+			return (X) this;
+		}
+
+		public String call() throws Exception {
+			this.chain(7).chain(5);
+			return "" + this;
+		}
+
+		@Override
+		public String toString() {
+			return "ParentHasDslMethodCall [i=" + i + "]";
+		}
+
+	}
+
+	static class ChildHasDslMethodCall extends ParentHasDslMethodCall<ChildHasDslMethodCall> implements Callable<String> {
+
+		public String call() throws Exception {
+			this.chain(1).chain(3);
+			return "" + this;
+		}
+
+	}
+    
+	@Before
+	public void setupEngineToRemoveVoidMethods() {
+		createTesteeWith(mutateOnlyCallMethod(), DSLMethodCallMutator.DSL_METHOD_CALL_MUTATOR);
+	}
+
+	@Test
+	public void shouldRemoveNonVoidMethods() throws Exception {
+		assertTrue(findMutationsFor(HasIntMethodCall.class).isEmpty());
+	}
+
+	@Test
+	public void shouldNotRemoveVoidMethodCalls() throws Exception {
+		assertTrue(findMutationsFor(HasVoidMethodCall.class).isEmpty());
+	}
+
+	@Test
+	public void shouldNotRemoveConstructorCalls() throws Exception {
+		final FunctionalList<MutationDetails> actual = findMutationsFor(HasConstructorCall.class);
+		assertFalse(actual.contains(descriptionContaining("Integer")));
+	}
+
+	@Test
+	public void shouldRemoveDslMethods() throws Exception {
+		FunctionalList<MutationDetails> methods = findMutationsFor(HasDslMethodCall.class);
+		assertEquals(1, methods.size());
+
+		final Mutant mutant = getFirstMutant(HasDslMethodCall.class);
+		assertMutantCallableReturns(new HasDslMethodCall(), mutant, "HasDslMethodCall [i=3]");
+	}
+
+
+	@Test
+	@Ignore
```

Comment:
FWIW, my bytecode manipulation knowledge is too limited to get this one working, but I appreciate any help!